cloudinit/config/cc_ssh.py

```diff
@@ -141,19 +139,11 @@
               rsa_public: ssh-rsa AAAAB3NzaC1yc2EAAAABIwAAAGEAoPRhIfLvedSDKw7Xd ...
               rsa_certificate: |
                 ssh-rsa-cert-v01@openssh.com AAAAIHNzaC1lZDI1NTE5LWNlcnQt ...
-              dsa_private: |
```

Comment:
security review: Do we want to fall back here if the host itself has openssl is built with `enable-weak-ssl-ciphers`? Could we check this with `openssl ciphers -s DSA` ? Or are we okay stating folks are off-plan on stable Ubuntu 20.04 if they still rely on DSA . and they'd need to generate their own host-keys with either supplemental `#cloud-config`  via either [write_files](https://cloudinit.readthedocs.io/en/latest/reference/modules.html#write-files) or [runcmd](https://cloudinit.readthedocs.io/en/latest/reference/modules.html#runcmd) customization

Reply:
I don't think that a fallback is necessary. Ubuntu has disabled support for ssh-dss in 16.04, so there aren't a whole lot of clients around that would work with this key and those that do probably shouldn't have network access.